engine/src/system_factory.cpp

```diff
@@ -331,7 +331,7 @@ Planet *SystemFactory::processPlanet(Star_XML *xml, Object &object, Planet *owne
         stream >> index >> local;
 
         Light light = lights[index];
-        GFXLight gfx_light(true, Vector(0, 0, 0));
+        GFXLight gfx_light(true, GFXColor(Vector(0, 0, 0), 1));
```

Comment:
This is an abomination.
Recycling Vector because it's a three value struct.
We should remove this from GFXColor and make the code here:

`GFXLight gfx_light(true, GFXColor(0, 0, 0, 1));`

Reply:
You're right. I should have caught that.

Reply:
OK. It took some doing, but I think I have now fixed this. (I had to add overloads of several methods in different classes so that they could take either a Vector& or a GFXColor&.)

Reply:
And that in itself is not ideal, but there doesn't seem to be a way around it at the moment.